shared/src/business/entities/contacts/ContactFactory.js

```
@@ -172,111 +166,129 @@ const getContactConstructor = ({
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPartnershipAsTaxMattersPartnerPrimaryContact,
+          secondary: null,
+        };
+      case PARTY_TYPES.partnershipBBA:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPartnershipBBAPrimaryContact,
+          secondary: null,
+        };
+      case PARTY_TYPES.partnershipOtherThanTaxMatters:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPartnershipOtherThanTaxMattersPrimaryContact,
+          secondary: null,
+        };
+      case PARTY_TYPES.petitioner:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPetitionerPrimaryContact,
+          secondary: null,
+        };
+      case PARTY_TYPES.petitionerDeceasedSpouse:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPetitionerPrimaryContact,
+          secondary: getPetitionerDeceasedSpouseContact,
+        };
+      case PARTY_TYPES.petitionerSpouse:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPetitionerPrimaryContact,
+          secondary: getPetitionerSpouseContact,
+        };
+      case PARTY_TYPES.survivingSpouse:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getSurvivingSpouseContact,
+          secondary: null,
+        };
+      case PARTY_TYPES.transferee:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPetitionerPrimaryContact,
+          secondary: null,
+        };
+
+      case PARTY_TYPES.trust:
+        return {
+          otherPetitioners: getOtherPetitionerContact,
+          primary: getPetitionerTrustContact,
+          secondary: null,
+        };
+      default:
+        if (partyType) {
+          console.warn(`Unrecognized party type "${partyType}"`);
```

Would we need to throw some kind of error at this point, or would a console.warn be enough? Asking because I am genuinely not sure!

---

We may need to just throw an error here instead, so it bubbles up to HB

---

The previous implementation would silently return "nothing", so I'll have to tweak some tests, but can-do.